virttest/qemu_devices/qdevices.py

```diff
@@ -1160,9 +1160,11 @@ def _cmdline_json(self):
                 command_dict[key] = str(val)
             # disable-legacy from device ("driver": "virtio-scsi-pci")
             # write-cache from device ("driver": "scsi-hd")
-            elif val in ('on', 'yes') and key not in expect_string_val:
+            # disable-modern from device ("driver": "virtio-scsi-pci")
```

Comment:
According to the other options `disable-legacy` and `write-cache`, we saw there is a variable `expect_string_val` to handle such behaviors that we don't change the value to the boolean type.
But for the options `disable-legacy` and `disable-modern`, the type of value seems like same from the raw command line, for example, `scsi=on,disable-legacy=off,disable-modern=on`.
Here we have different ways to handle the options `disable-legacy` and `disable-modern` in the json command line.
I don't know if it is a difference between `disable-legacy` and `disable-modern` to handle the value in the json format from the qemu product side.
If it is, I suggest we may remove the comment `# disable-modern from device ("driver": "virtio-scsi-pci")` here to avoid some mumbling.

Reply:
> According to the other options `disable-legacy` and `write-cache`, we saw there is a variable `expect_string_val` to handle such behaviors that we don't change the value to the boolean type. But for the options `disable-legacy` and `disable-modern`, the type of value seems like same from the raw command line, for example, `scsi=on,disable-legacy=off,disable-modern=on`. Here we have different ways to handle the options `disable-legacy` and `disable-modern` in the json command line. I don't know if it is a difference between `disable-legacy` and `disable-modern` to handle the value in the json format from the qemu product side. If it is, I suggest we may remove the comment `# disable-modern from device ("driver": "virtio-scsi-pci")` here to avoid some mumbling.

The qemu command line exists: `-device '{"id": "virtio_scsi_pci0", "driver": "virtio-scsi-pci", "bus": "pcie-root-port-2", "addr": "0x0", "disable-legacy": "on", "disable-modern": false}' `
@YongxueHong , the comment has been deleted. Thanks.